multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineLoginFilter.java

```diff
@@ -70,12 +71,29 @@ public MachineLoginFilter(
       PermissionChecker permissionChecker) {
     this.tokenExtractor = tokenExtractor;
     this.userManager = userManager;
-    this.keyManager = keyManager;
     this.permissionChecker = permissionChecker;
+    this.jwtParser =
```

Comment:
This field initialization looks quite heavy. I think it worth to inject it from a container. With a provider for example.